common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -212,10 +212,13 @@ def delete_item(self, location, user_id, revision=None, **kwargs):
                 branched_location = location.for_branch(branch)
                 parent_loc = self.get_parent_location(branched_location)
                 SplitMongoModuleStore.delete_item(self, branched_location, user_id)
-                # publish parent w/o child if deleted element is direct only (not based on type of parent)
+                # publish parent w/o child if deleted element is direct only (not based on type of parent) or vertical
                 if (
                         branch == ModuleStoreEnum.BranchName.draft and
-                        branched_location.block_type in DIRECT_ONLY_CATEGORIES and parent_loc
+                        (
+                            branched_location.block_type in DIRECT_ONLY_CATEGORIES or
+                            branched_location.block_type == 'vertical'
```

Comment:
Does calling `block_type` on `branched_location` add more mongo queries? If so, assign the value to a variable and use the variable in L219-220